src/docopt/dispatch.nim

```diff
@@ -0,0 +1,59 @@
+import docopt, macros, strutils, sequtils
+
+proc to[T](v: Value): T =
+  when T is SomeInteger:
+    T(parseInt($v))
+  elif T is bool:
+    v.toBool
+  elif T is string:
+    $v
+  elif T is seq:
+    @v
```

Comment:
What happens else?

Reply:
I think the idea is that these are the only types covered by the `Value` variant object.

I agree though that either `else` should be supplied and given a `{.error: "Invalid type ...".} kind of message or the generic should be restricted to the possible types. The latter is cleaner, imo.

Reply:
Definitely should be an error there. I will add one.